### ISSUE TITLE
Question: Is it possible to turn json back into replay file 

### ISSUE THREAD

**Issue by markduk** [jjbott/RocketLeagueReplayParser#44]
I'm interested to know if it is possible to turn the json back into a replay file?  This would open a whole world of editing replays that can then be viewed via RocketLeague.
- Change player names
- Change map
- Change car

**Comment by tfausak** [comment]
Hopefully @jjbott won't mind me shilling my own project here :) 

[Rattletrap](https://github.com/tfausak/rattletrap) can do that.

**Comment by jjbott** [comment]
Yeah, I never implemented that. In theory you could use my library to fiddle with the deserialized replay in memory, and then re-serialize it.

**Comment by markduk** [comment]
I see the Serialize method in the library, does it work?  If so I could easily extend the console to convert from .json to .replay.

**Comment by jjbott** [comment]
Sorry I'm slow. Yeah, in theory `Serialize` works to output a replay. 

I don't remember why deserializing from JSON was tough. Deserializing from "Pretty" JSON might be tricky, since it does some data mangling to generate it. "Raw" JSON might not be so bad. Could just be that no one uses my library, so no one asked for it, so why bother. 😄

**Comment by Gutbuster** [comment]
It's being used. But I think altering the replays is a slightly obscure need. I also looked into it at some point out of curiosity. Can't remember why. Maybe for adding some debug strings or playing a prank. Can't exactly recall.

**Comment by jjbott** [comment]
Yeah, I've always thought it'd be fun to create a completely custom replay from scratch. But unless you're doing something simple like changing someone's name in an existing replay, it's _crazy_ complicated. Like, if you want to inject a player and have it move around, it's possible but you wont have a fun time. :)

**Comment by markduk** [comment]
I'm interested to hear how you and others use this library and why you wrote it in the first place.  What problem were you trying to solve?  Are you trying to extract data from the replay files for some reason? I can see value in doing that.

**Comment by jjbott** [comment]
I personally don't use it. I don't even play Rocket League anymore. 😄 But keeping the library up to date is a fun challenge.

Ages ago I thought it'd be interesting to see what kind of stats you could generate if you could parse the replays. There was a [thread on the old Rocket League Forums](https://web.archive.org/web/20190501232510/https://psyonix.com/forum/viewtopic.php?f=33&t=13656) where people were trying to figure out how to parse them, with a tiny bit of dev help. I joined that, and it was a lot of fun. At the end it felt like a race to see who would get a working parse first. 😄 

Once I was able to parse them, I tried generating some visualizations for them. Like, there is code for generating heatmaps in the git history somewhere. But I found them useless, and the "race" was over, so I mostly lost interest. 

Long story short, I ultimately never generated anything useful from this. I hope someone out there is!

**Comment by somecsharp** [comment]
I think it's pretty cool and appreciate your hard work on this. Has anyone ever tried to persist a replay to a relational database? Much easier to manipulate or analyze after that. The biz objects and models look merged here but easy to separate out. I may give it a shot for fun...
